connector/src/main/java/org/geysermc/connector/entity/Entity.java

```diff
@@ -166,14 +182,52 @@ public void moveRelative(GeyserSession session, double relX, double relY, double
     }
 
     public void moveRelative(GeyserSession session, double relX, double relY, double relZ, Vector3f rotation, boolean isOnGround) {
+        Position bedLocation = new Position((int) position.getFloorX(), (int) position.getFloorY(), (int) position.getFloorZ());
+        bedId = session.getConnector().getWorldManager().getBlockAt(session, bedLocation);
```

Comment:
Chunk caching may be off - add a check to see if chunk caching is enabled: `if (session.getConnector().getConfig().isCacheChunks()`...

Reply:
(And if it isn't you'll have to skip over your bed position checking)

Reply:
Thanks guys I will make those changes